packages/provider/src/KnownRelaysManager.ts

```diff
@@ -80,7 +80,16 @@ export class KnownRelaysManager {
 
     this.logger.info(`fetchRelaysAdded: found ${relayInfos.length} relays`)
 
-    const filteredRelayInfos = relayInfos.filter(this.relayFilter)
+    const blacklistFilteredRelayInfos = relayInfos.filter((info: RelayRegisteredEventInfo) => {
+      let host = ''
+      try {
+        host = new URL(info.relayUrl).host.toLowerCase()
+      } catch (e) {}
+      const isHostBlacklisted = this.config.blacklistedRelays.find(relay => host.includes(relay.toLowerCase())) != null
```

Comment:
better match against relayUrl, and not just the "host" part.
that is, allow filtering by full url, including the "https" prefix
(e.g. copying a URL from the preferred relay into the blacklistedRelays should work)

not sure if the blacklist-by-address is useful, but it causes no harm either.

Reply:
```js
 const isHostBlacklisted = this.config.blacklistedRelays.find(relay => info.relayUrl.toLowerCase().includes(relay.toLowerCase())) != null
```